content/ui-patterns/saving.mdx

```diff
@@ -0,0 +1,266 @@
+
+- Text inputs
+- Checkbox groups
+- Radio button groups
+- Browser-native `<select>` dropdown menus
+- Multi-select dropdown menus
+
+To ensure your forms are accessible, do not automatically save declarative inputs.
+
+There are a few issues to bear in mind with autosaving declarative inputs.
+
+- Text inputs: Users expect to set a value and then submit. Sensitive information could be unintentionally submitted, such as a change password input.
+- Checkbox groups: It could be unclear whether or not the selections has been saved. Users may accidentally submit a selection clicked by accident.
+- Radio button groups: Screenreader users can't read the options without selecting them, which could accidentally apply a selection the user didn't want.
+- Browser-native `<select>` dropdowns: Similarly to radio button groups, options could be selected when a user focuses the select and uses the arrow keys to read through each option.
+
+<DoDontContainer>
+  <Do>
+    <img src="https://user-images.githubusercontent.com/2313998/151623711-ad294946-fd67-428d-96b2-e4f534857c8a.png" />
+    <Caption>Use save and cancel buttons in dropdowns that allow multiple selections</Caption>
+  </Do>
+  <Dont>
+    <img src="https://user-images.githubusercontent.com/2313998/151623713-12028e6d-86ba-4ae6-b46b-d3c88c0fa2c0.png" />
+    <Caption>Don't automatically save selections once the dropdown is closed</Caption>
+  </Dont>
+</DoDontContainer>
+
+## Automatic saving
+
+Automatic saving applies changes as they are made. Automatic saving should be used when the user expects instant feedback from the change they made.
+
+<Box mb={3} display="flex" alignItems="center" flexDirection="column" mt={6} mx="auto" sx={{gap: 2, maxWidth: '450px'}}>
+    <Box display="inline-block" overflow="hidden" borderWidth="1px" borderColor="border.default" borderStyle="solid" borderRadius="12px">
+        <img
+            width="456"
+            src="https://user-images.githubusercontent.com/2313998/152013139-02eb5213-bb19-4c8d-8b19-01f1d73602d1.gif"
+            alt="Opening the 'Watch' dropdown menu and changing. It saves automatically."
+        />
+    </Box>
+    <Text as="p" mt="0" align="center" fontSize={1}>If a form field is saved automatically, it should be obvious whether or not it saved without using text or visual indicators. A visual indicator for whether or not a form field has been saved successfully could be mistaken as a validation status.</Text>
+</Box>
+
+### Dropdown menu accessibility
+
+Automatically saving a single-select dropdown menu is accessible as long as it is a semantic menu. A semantic menu can only be a list of items that can be selected.
+
+As soon as a dropdown has additional features (search inputs, filters, tabs, etc), the dropdown should be treated like a dialog with a button to submit and a button to cancel. A dialog will provide the proper navigation features for folks using assistive technology.
```

Comment:
```suggestion
### Dropdown menu accessibility

Semantic menus should only have a list of options to select from and should not have a save button in their dropdown. As soon as additional features (search inputs, filters, tabs, etc) are added, the component should be treated like a dialog with a button to submit and a button to cancel. 
```

Trimmed it a bit; let me know what you think @mperrotti. I took out the last section about screen readers because it provides a better experience to more than just assistive technology users 🙂